### PR TITLE
Refactor test to use stretchr/testify

### PR DESCRIPTION
**Pull request by roidelapluie**

### CODE REVIEW

File: discovery/eureka/eureka_test.go [hunk 1]
```diff
@@ -45,7 +46,7 @@ func testUpdateServices(respHandler http.HandlerFunc) ([]*targetgroup.Group, err
 
 func TestEurekaSDHandleError(t *testing.T) {
 	var (
-		errTesting  = errors.Errorf("non 2xx status '%d' response during eureka service discovery", http.StatusInternalServerError)
+		errTesting  = fmt.Sprintf("non 2xx status '%d' response during eureka service discovery", http.StatusInternalServerError)
```
Comment (brian-brazil) [inline discovery/eureka/eureka_test.go:49]: Couldn't we hardcode here? This always produces the same value.

File: discovery/hetzner/robot_test.go [hunk 2]
```diff
@@ -42,16 +44,16 @@ func TestRobotSDRefresh(t *testing.T) {
...
-	testutil.Ok(t, err)
-	testutil.Equals(t, 1, len(targetGroups))
+	assert.NoError(t, err)
+	assert.Equal(t, 1, len(targetGroups))
 
 	targetGroup := targetGroups[0]
-	testutil.Assert(t, targetGroup != nil, "targetGroup should not be nil")
-	testutil.Assert(t, targetGroup.Targets != nil, "targetGroup.targets should not be nil")
-	testutil.Equals(t, 2, len(targetGroup.Targets))
+	assert.True(t, targetGroup != nil, "targetGroup should not be nil")
```
Comment (brian-brazil) [inline discovery/hetzner/robot_test.go:54]: There's a NotNil

File: go.sum [hunk 3]
```diff
@@ -770,6 +770,8 @@ github.com/stretchr/testify v1.3.0/go.mod h1:M5WIy9Dh21IEIfnGCwXGc5bZfKNJtfHm1UV
 github.com/stretchr/testify v1.4.0/go.mod h1:j7eGeouHqKxXV5pUuKE4zz7dFj8WfuZ+81PSLYec5m4=
```
Comment (brian-brazil) [inline go.sum:770]: Based on the go.mod changes, I'd have expected some lines to be removed here.

File: rules/manager_test.go [hunk 4]
```diff
@@ -24,11 +24,14 @@ import (
 	"time"
 
 	"github.com/go-kit/kit/log"
-	"github.com/prometheus/client_golang/prometheus"
-	"github.com/prometheus/common/model"
 	"go.uber.org/goleak"
 	yaml "gopkg.in/yaml.v2"
 
+	"github.com/prometheus/client_golang/prometheus"
```
Comment (brian-brazil) [inline rules/manager_test.go:30]: Why are these separate? They're all 3rd party libraries from an import standpoint.

File: storage/remote/queue_manager_test.go [hunk 5]
```diff
@@ -32,6 +31,10 @@ import (
 	"github.com/golang/snappy"
 	"go.uber.org/atomic"
 
+	"net/url"
```
Comment (brian-brazil) [inline storage/remote/queue_manager_test.go:34]: This is internal, it belongs with the other internals.

**Review by brian-brazil — COMMENTED** [review]
The changes made in passing to imports don't seem right.

Good to see our code simplified and cleaned up.

**Comment by roidelapluie** [comment]
Thanks, I have addressed  + fixed more tests

File: go.sum (hunk 3 above)
Comment (roidelapluie) [reply go.sum:746]: I have removed go.sum, removed my local go cache, and run go mod tidy, I can't do better than my last iteration.

File: cmd/prometheus/main_test.go [hunk 6]
```diff
@@ -26,12 +26,13 @@ import (
 	"time"
 
 	"github.com/go-kit/kit/log"
+	"github.com/stretchr/testify/assert"
+
```
Comment (brian-brazil) [inline cmd/prometheus/main_test.go:30]: Why are you adding a blank line, here and elsewhere?

Comment (brian-brazil) [reply cmd/prometheus/main_test.go:30]: Please do, prometheus/prometheus is what's local here.

Comment (roidelapluie) [reply cmd/prometheus/main_test.go]: The last commit refactors the whole repo imports, with prometheus/prometheus as local.

**Comment by brian-brazil** [comment]
Looks fine, but you've conflicts.

**Comment by roidelapluie** [comment]
Conflicts are fixed

**Comment by brian-brazil** [comment]
:+1:

**Comment by bwplotka** [comment]
Why it was done? (: 

I disagree with this. We now pull so many unnecessary deps only to get rid of few lines of code and have worse too wide API in our tests. This feels wrong ):

**Comment by roidelapluie** [comment]
> Why it was done? (:
> 
> I disagree with this. We now pull so many unnecessary deps only to get rid of few lines of code and have worse too wide API in our tests. This feels wrong ):
> 
> The API is awful because now you spent a lot more time to decide `should it be required.XYZ or assert.XYZ.` and many more like this. Same when reading and reviewing.

We never use `require`.

Originally because our testutil.Equal was not showing private fields, I wanted to refactor it but it turns out that the one from the library picked did fix my needs.

Regarding the choice of the library: I picked this library because cortex is also using it.

**Comment by bwplotka** [comment]
Unfortunately, it's not that simple. Now in most of the tests, when they fail, `assert` flow will go further and cause harder to read panics, because it's usually the case that we check something which is expected non-nil when error is non-nil. 

That's why if anything we should probably use `require` only. Thus we are at the same, bad spot.

> Regarding the choice of the library: I picked this library because cortex is also using it.

Hm, @codesome just mentioned he had refactored tons of code to testify after this change ): I guess partially was using. Would be nice to agree on a limited form `testutil` of our choice. Especially for so widely used package - it has to be limited.

Anyway, at the current version (assert) our tests will panic on most of the failures, sorry for late review.

**Comment by roidelapluie** [comment]
You are right, we should use require. testutil was using Fatalf.

**Comment by roidelapluie** [comment]
> Unfortunately, it's not that simple. Now in most of the tests, when they fail, `assert` flow will go further and cause harder to read panics, because it's usually the case that we check something which is expected non-nil when error is non-nil.
> 
> That's why if anything we should probably use `require` only. Thus we are at the same, bad spot.
> 
> > Regarding the choice of the library: I picked this library because cortex is also using it.
> 
> Hm, @codesome just mentioned he had refactored tons of code to testify after this change ): I guess partially was using. Would be nice to agree on a limited form `testutil` of our choice. Especially for so widely used package - it has to be minimal.
> 
> Anyway, at the current version (assert) our tests will panic on most of the failures, sorry for late review.

created #8122

We still have testutil. As said, our Equal function was limited, testify does it better. If we decide to reintroduce more test functions, we should do it in common so other components can reuse them.

**Comment by bwplotka** [comment]
Thanks!

Agree, thanks for clarifications :hugs:

**Comment by roidelapluie** [comment]
:hugs:

**Pull request merged**